modules/data_release/php/data_release.class.inc

```diff
@@ -51,70 +54,83 @@ class Data_Release extends \NDB_Menu_Filter
-                               'file_name',
-                               'version',
-                               'upload_date',
-                              );
-        $this->validFilters = array(
-                               'file_name',
-                               'version',
-                               'upload_date',
-                              );
-        $this->formToFilter = array(
-                               'file_name'   => 'file_name',
-                               'version'     => 'version',
-                               'upload_date' => 'upload_date',
-                              );
-
-        $userUpload     = $user->hasPermission("data_release_upload");
-        $userEditAccess = $user->hasPermission("data_release_edit_file_access");
+        $this->group_by = '';
+        $this->order_by = 'uploadDate';
+    }
 
-        $userids = $DB->pselectColWithIndexKey(
-            "SELECT ID, UserID FROM users",
-            array(),
-            "ID"
-        );
+    /**
+     * Create the form for the data release menu page
+     *
+     * @return void
+     **/
+    function setup()
+    {
+        parent::setup();
 
-        $versionedFiles        = $this->getVersionedFiles($DB);
-        $data_release_ids      = array();
-        $data_release_versions = array();
+        // Get list of users in the database
+        $list_of_users = $this->getUsersList();
 
-        foreach ($versionedFiles as $versionName=>$versionFiles) {
-            foreach ($versionFiles as $fileID=>$fileName) {
-                $data_release_ids[$fileID] = $fileName . " - " . $versionName;
-            }
-            $data_release_versions[$versionName] = $versionName;
+        // Get list of data release versions and filenames
+        $vFiles = $this->getVersionedFiles();
 
+        $filenames_options = array();
+        $versions_options  = array();
+
+        foreach ($vFiles as $versionName => $filesArray) {
+            foreach ($filesArray as $fileID => $fileName) {
+                $filenames_options[$fileID] = "$fileName - $versionName";
+            }
+            $versions_options[$versionName] = $versionName;
         }
 
-        $manage_permissions = $this->getUserVersionPermissions($versionedFiles, $DB);
+        // Get list of permissions per data release versions
+        $manage_permissions = $this->getUserVersionPermissions($vFiles);
+
+        $this->fieldOptions = [
+                               'users'              => $list_of_users,
+                               'versions'           => $versions_options,
+                               'filenames'          => $filenames_options,
+                               'manage_permissions' => $manage_permissions,
+                              ];
+    }
+
+
+    /**
+     * Greps the list of users available in the users database table.
+     *
+     * @return array Array of user ID and user name
+     */
+    function getUsersList()
```

Comment:
This should also take a \Database as a parameter instead of using NDB_Factory

Reply:
done